### PR TITLE
Crosschain

### PR DESCRIPTION
**Pull request by AlexNi245**

### CODE REVIEW

File: packages/messenger-widget/src/components/Contacts/bl.ts [hunk 1]
```diff
@@ -25,6 +25,8 @@ import { getDeliveryServiceProperties } from 'dm3-lib-delivery-api';
 import { MessageState } from 'dm3-lib-messaging';
 import axios from 'axios';
 import { globalConfig } from 'dm3-lib-shared';
+import { mainnet } from 'wagmi';
+import e from 'cors';
```
Comment (hai-ko) [inline packages/messenger-widget/src/components/Contacts/bl.ts:29]: Is that needed?

File: packages/messenger-widget/src/components/DM3/DM3.tsx [hunk 2]
```diff
@@ -52,33 +58,32 @@ function DM3(props: Dm3Props) {
...
+    }, [account?.profile]);
 
     // handles socket connection &  contacts with message fetching
     useEffect(() => {
-        if (
-            state.connection.connectionState === ConnectionState.SignedIn &&
-            !state.connection.socket &&
-            deliveryServiceUrl
-        ) {
+        console.log('start soccect connect efffect');
```
Comment (hai-ko) [inline packages/messenger-widget/src/components/DM3/DM3.tsx:77]: console.log should be removed

File: packages/messenger-widget/src/hooks/auth/AccountConnector.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                    address,
+                    process.env.REACT_APP_RESOLVER_BACKEND as string,
+                )) ?? getAliasForAddress(address);
+
+            //We're trying to get the profile from the delivery service
+            const userProfile = await getUserProfile(mainnetProvider, ensName);
+
+            return { ensName, userProfile };
+        } catch (e) {
+            console.log('error ', e);
```
Comment (hai-ko) [inline packages/messenger-widget/src/hooks/auth/AccountConnector.ts:35]: That should be removed

File: packages/messenger-widget/src/hooks/auth/DeliveryServiceConnector.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        if (!address) {
+            throw Error('No eth address');
+        }
+
+        const storageKeyCreationMessage = getStorageKeyCreationMessage(
+            nonce,
+            address,
+        );
+
+        //@ts-ignore
```
Comment (hai-ko) [inline packages/messenger-widget/src/hooks/auth/DeliveryServiceConnector.ts:56]: Why //@ts-ignore ?

File: packages/lib/shared/package.json [hunk 5]
```diff
@@ -8,7 +8,11 @@
         ".": {
             "import": "./dist/index.js",
             "require": "./dist/index.js"
-        }
+        },
+        "./dist/ethersHelper": {
```
[the remark this answers is not in the document]
Comment (AlexNi245) [reply packages/lib/shared/package.json:12]: FE build fails when ethersHelper is not exported. I dont know why, maybe we havent used ethers helper in the FE so far

**Review by hai-ko — APPROVED** [review]

**Pull request merged**